### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by maks45**

### CODE REVIEW

File: src/main/java/core/basesyntax/FieldXY.java [hunk 1]
```diff
@@ -13,7 +13,32 @@
...
+            robot.turnLeft();
+        }
+    }
+
+    private void setVerticalDirection(Robot robot, int toY) {
+        Direction neededDirection = robot.getX() < toY ? Direction.UP : Direction.DOWN;
+        while (!robot.getDirection().equals(neededDirection)) {
+            robot.turnLeft();
+        }
     }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java]: can we combine these methods into one?

File: src/main/java/core/basesyntax/FieldXY.java [hunk 2]
```diff
@@ -13,7 +13,32 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() != toX) {
+            setHorizontalDirection(robot, toX);
+            while (robot.getX() != toX) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java]: we are checking the same condition twice

File: src/main/java/core/basesyntax/FieldXY.java [hunk 3]
```diff
@@ -13,7 +13,32 @@
...
+                robot.stepForward();
+            }
+        }
+        if (robot.getY() != toY) {
+            setVerticalDirection(robot, toY);
+            while (robot.getY() != toY) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java]: same

File: src/main/java/core/basesyntax/FieldXY.java (hunk 1 above)
Comment (maks45) [reply src/main/java/core/basesyntax/FieldXY.java]: I combined these two methods into one with switch -case

File: src/main/java/core/basesyntax/FieldXY.java (hunk 2 above)
Comment (maks45) [reply src/main/java/core/basesyntax/FieldXY.java]: got it

File: src/main/java/core/basesyntax/FieldXY.java (hunk 3 above)
Comment (maks45) [reply src/main/java/core/basesyntax/FieldXY.java]: got it

File: src/main/java/core/basesyntax/FieldXY.java [hunk 4]
```diff
@@ -13,7 +13,68 @@
...
+        neededDirection = robot.getY() < toY ? Direction.UP : Direction.DOWN;
+        if (!robot.getDirection().equals(neededDirection)) {
+            setNeededDirection(robot, neededDirection);
+        }
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
+    }
+
+    private void setNeededDirection(Robot robot, Direction neededDirection) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:32]: try to optimize this method it's too long

Comment (maks45) [reply src/main/java/core/basesyntax/FieldXY.java:27]: done

**Pull request closed without merging**